google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingHandler.java

```diff
@@ -321,7 +324,15 @@ public void publish(LogRecord record) {
         if (redirectToStdout) {
           logEntries.forEach(log -> System.out.println(log.toStructuredJsonString()));
         } else {
-          logging.write(logEntries, defaultWriteOptions);
+          // Add partialSuccess option always for request containing instrumentation data
+          if (pair.x()) {
+            List<WriteOption> writeOptions = new ArrayList<WriteOption>();
+            writeOptions.addAll(Arrays.asList(defaultWriteOptions));
+            writeOptions.add(WriteOption.partialSuccess(true));
+            logging.write(logEntries, Iterables.toArray(writeOptions, WriteOption.class));
+          } else {
+            logging.write(logEntries, defaultWriteOptions);
+          }
```

Comment:
nit: the flag can always be added to avoid if/else: `WriteOption.partialSuccess(pair.x())`

Reply:
I reworked this part of code - let me know WDYT. I was trying to avoid adding partialSuccess always to make sure we do not break existing tests